go/master/service_test.go

```diff
@@ -0,0 +1,69 @@
+package master_test
+
+import (
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/PaddlePaddle/Paddle/go/master"
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/embed"
+	"github.com/docker/docker/pkg/ioutils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewServiceWithEtcd(t *testing.T) {
+	// setup an embed etcd server
+	etcdDir, err := ioutils.TempDir("", "")
+	if err != nil {
+		t.Fatal(nil)
+	}
+	cfg := embed.NewConfig()
+	cfg.Dir = etcdDir
+	e, err := embed.StartEtcd(cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		e.Close()
+		if err := os.RemoveAll(etcdDir); err != nil {
+			log.Fatal(err)
+		}
+	}()
+	select {
+	case <-e.Server.ReadyNotify():
+		log.Printf("Server is ready!")
```

Comment:
Maybe `t.Log` would be better (prints only when test failed or `-v` is specified). Same for other places.